src/configureWorkspace/configureCompose.ts

```diff
@@ -0,0 +1,109 @@
+        return captureCancelStep(step, context.telemetry.properties, prompt);
+    }
+
+    const serviceNames: string[] = [];
+    let hasDuplicateService: boolean = false;
+    function AddToServiceNames(serviceName: string): void {
+        if (serviceNames.includes(serviceName)) {
+            hasDuplicateService = true;
+        } else {
+            serviceNames.push(serviceName);
+        }
+    }
+
+    const rootFolder: vscode.WorkspaceFolder = await captureStep('folder', promptForFolder)();
+    let composeFile: ScaffoldFile;
+    // Get list of dockerfiles (services) to add to compose
+    const dockerFiles: Item[] = await getDockerFilesInWorkspace(context, rootFolder);
+
+    // Add the services to docker-compose.yaml
+    if (dockerFiles) {
+        const composeScaffoldContexts: ComposeScaffoldContext[] = [];
+        await Promise.all(
+            dockerFiles.map(async dockerfile => {
+                const serviceName = path.basename(path.dirname(dockerfile.absoluteFilePath));
+                const dockerfileInfo: DockerfileInfo = await parseDockerfile(rootFolder.uri.fsPath, dockerfile.absoluteFilePath);
+                AddToServiceNames(serviceName);
+                composeScaffoldContexts.push({
+                    ...dockerfileInfo,
+                    serviceName: serviceName,
+                });
+            })
+        );
+        if (hasDuplicateService) {
+            updateWithUniqueServiceName(composeScaffoldContexts, serviceNames);
+        }
+        composeFile = await generateComposeFiles(composeScaffoldContexts);
+    } else {
+        // No dockerfile is present in the workspace. Create a template docker-compose file.
+        const templateComposeFile = path.join(ext.context.asAbsolutePath('resources'), 'template.docker-compose.yml');
+        const composeContent = (await fse.readFile(templateComposeFile)).toString();
+        composeFile = { fileName: 'docker-compose.yml', contents: composeContent }
+    }
+    composeFile.onConflict = async (filePath) => { return await generateNonConflictFileNameWithPrompt(filePath) };
+    await writeFiles([composeFile], rootFolder.uri.fsPath);
+}
+
+function updateWithUniqueServiceName(contexts: ComposeScaffoldContext[], existingNames: string[]): void {
+    const processedServices: string[] = [];
+    contexts.map(context => {
+        if (processedServices.includes(context.serviceName)) {
+            context.serviceName = generateUniqueName(context.serviceName, existingNames, i => `.${i}`);
+            existingNames.push(context.serviceName);
+        }
+        processedServices.push(context.serviceName);
+    })
+}
+
+async function promptForFolder(): Promise<vscode.WorkspaceFolder> {
+    return await quickPickWorkspaceFolder(localize('vscode-docker.scaffolding.dockerCompose', 'To generate docker-compose files you must first open a folder or workspace in VS Code.'));
```

Comment:
To make duplicate IDs less likely:
```suggestion
    return await quickPickWorkspaceFolder(localize('vscode-docker.scaffolding.dockerCompose.noWorkspaceFolder', 'To generate docker-compose files you must first open a folder or workspace in VS Code.'));
```